marge/app.py

```diff
@@ -27,138 +28,153 @@ def time_interval(s):
         raise argparse.ArgumentTypeError('Invalid time interval (e.g. 12[s|min|h]): %s', s)
 
 
-def _parse_args(args):
-    parser = argparse.ArgumentParser(description=__doc__)
-
-    def arg(name, env_var=True, **kwargs):
-        if env_var:
-            env_var_name = 'MARGE_' + name[2:].replace('-', '_').upper()
-            env_var_val = os.getenv(env_var_name)
-            if env_var_val:
-                if kwargs.get('action') in ['store_true', 'store_false']:
-                    env_var_val = env_var_val.lower() in ['true', 'yes', '1']
-                kwargs = dict(kwargs, default=env_var_val, required=False)
+def _parse_config(args):
 
-        parser.add_argument(name, **kwargs)
+    for bad_arg in ['--auth-token', '--ssh-key']:
+        if bad_arg in args:
+            raise Exception('"%s" can only be set via ENV var or config file.' % bad_arg)
 
     def regexp(s):
         try:
             return re.compile(s)
         except re.error as err:
             raise argparse.ArgumentTypeError('Invalid regexp: %r (%s)' % (s, err.msg))
 
-    arg(
+    parser = configargparse.ArgParser(
+        auto_env_var_prefix='MARGE_',
+        ignore_unknown_config_file_keys=True,  # Don't parse unknown args
+        config_file_parser_class=configargparse.YAMLConfigFileParser,
+        args_for_setting_config_path=["--config-file"],
+        formatter_class=configargparse.ArgumentDefaultsRawHelpFormatter,
+        description=__doc__,
+    )
+    auth_token_group = parser.add_mutually_exclusive_group(required=True)
+    auth_token_group.add_argument(
+        '--auth-token',
+        type=str,
+        metavar='TOKEN',
+        help=(
+            'Your gitlab token.\n'
+            'WARNING:\n'
+            'Due to security reasons, you can only set this via ENV var or config file.\n'
```

Comment:
mention `--auth-token-file`?

Reply:
This is about '--auth-token', 'MARGE_AUTH_TOKEN' and 'auth-token' in config file. What i tried to say is use ENV var or config map.

I think '--auth-token-file' should be seen as a separate option?

Reply:
But it is related. If you want to use a commandline flag (vs config file or env var), you should use --auth-token-file instead of --auth-token. 

I'd write somethihng like "DISABLED because passing credentials on the command line is insecure; use the identical config file option, set MARGE_AUTH_TOKEN or instead specify a file with --auth-token-file'